fakes_test.go

```diff
@@ -24,16 +62,56 @@ func Test_VirtualMachinesClient_Get(t *testing.T) {
 
 	// the fake VM must return the provided name and its ID contain the provided resource group name.
 
-	// TODO: populate vm with response from fake
+	responseVM := armcompute.VirtualMachine{
+		Name: to.Ptr[string](vmName),
+		ID:   to.Ptr[string](resourceGroupName), // This isn't a real ID but good enough for our purposes
+	}
+
 	var vm armcompute.VirtualMachine
 
+	// matthchr: It wasn't clear to me until I read the README in armcompute that I needed to embed the fake into my own type,
+	// I had guessed at that but then though I was missing something because it's going to make things pretty verbose (need to define a new type for each test).
+	// Actually I also have to somehow pass the VM to return up to the server (or hardcode it) which is also unfortunate sorta.
+	// matthchr: Have you thought about using lambdas and doing this instead?
+	// fakeServer := &fake.VirtualMachinesServer{
```

Comment:
This is an interesting idea.  I'm going to experiment with this.

Reply:
This works way better and gives the best of both worlds.  Thanks for the suggestion!

Reply:
In theory you could support both ways if you wanted too, though maybe that gets sufficiently complicated that it's not worth it. Not sure. Definitely worth some thinking on. I can see cases where a module scoped single mock is cleaner, and other cases where it's not I think.

Reply:
We can get the single fake by declaring it in a global.  We can also decrease the number of types we need to generate which is a plus.
```go
var fakeSrv2 = computefake.VirtualMachinesServer2{
	BeginDelete: func(ctx context.Context, resourceGroupName, vmName string, options *armcompute.VirtualMachinesClientBeginDeleteOptions) (resp azfake.PollerResponder[armcompute.VirtualMachinesClientDeleteResponse], err azfake.ErrorResponder) {
		resp.AddNonTerminalResponse(nil)
		resp.SetTerminalError("ErrorOperationFailed", http.StatusConflict)
		return
	},
	NewListPager: func(resourceGroupName string, options *armcompute.VirtualMachinesClientListOptions) (resp azfake.PagerResponder[armcompute.VirtualMachinesClientListResponse]) {
		resp.AddPage(armcompute.VirtualMachinesClientListResponse{
			VirtualMachineListResult: armcompute.VirtualMachineListResult{
				Value: []*armcompute.VirtualMachine{
					&armcompute.VirtualMachine{},
					&armcompute.VirtualMachine{},
				},
			},
		}, nil)
		resp.AddError(errors.New("transient failure"))
		resp.AddPage(armcompute.VirtualMachinesClientListResponse{
			VirtualMachineListResult: armcompute.VirtualMachineListResult{
				Value: []*armcompute.VirtualMachine{
					&armcompute.VirtualMachine{},
					&armcompute.VirtualMachine{},
					&armcompute.VirtualMachine{},
				},
			},
		}, nil)
		return
	},
}
```